### DISCUSSION TITLE
GitHub Composite Actions - File Access

### DISCUSSION THREAD

**Discussion by Christian-Oleson** [community/community#49377] [Actions] [answered]
### Select Topic Area

Question

### Body

I am building a composite GitHub action that I want to have access to the files of the repos calling it. The idea is to package cloud resources for AWS Lambda and then upload them. However, I see this warning:

> We strongly recommend that actions use variables to access the filesystem rather than using hardcoded file paths. GitHub sets variables for actions to use in all runner environments.

I've tried to both change directories as well as set up the working directory, both of which do not succeed and do not find the code repo that calls the action.

What is the correct way to access the file system of a repo that calls this composite action? I do not want to pass things back because it will have to take a secret/key as a parameter in the action to allow it to access an AWS account.

**Comment by airtower-luna** [comment, marked as answer]
> What is the correct way to access the file system of a repo that calls this composite action?

The default working directory of the action will be the workspace of the job. To get the repository files the job should usually call [`actions/checkout`](https://github.com/actions/checkout) before calling your action.

The warning your quoted mostly means you shouldn't hard-code paths (beyond defaults), so people using your action can tell it where to find files, instead of having to follow a fixed file layout.

**Reply by Christian-Oleson** [reply]
This was the answer. The consumer for my composite action didn't check out their code first. Thanks!
